lnd.go

```diff
@@ -997,23 +1006,27 @@ func waitForWalletPassword(grpcEndpoints, restEndpoints []string,
 		)
 
 		// With the seed, we can now use the wallet loader to create
-		// the wallet, then unload it so it can be opened shortly
+		// the wallet, then pass it back to avoid unlocking it again.
 		birthday := cipherSeed.BirthdayTime()
-		_, err = loader.CreateNewWallet(
+		newWallet, err := loader.CreateNewWallet(
 			password, password, cipherSeed.Entropy[:], birthday,
 		)
 		if err != nil {
-			return nil, err
-		}
+			ltndLog.Errorf("Could not create new wallet: %v", err)
 
-		if err := loader.UnloadWallet(); err != nil {
+			// Don't leave the file open in case the new wallet
+			// could not be created for whatever reason.
+			if err := loader.UnloadWallet(); err != nil {
```

Comment:
Actually, a better pattern would be 
```go
if err != nil {
    if err := loader.UnloadWallet(); err != nil {
           log(err)
    }
    return nil, err
}
```
as it's the first error that causes us to fail in the first place, and makes sense to return 😛

Reply:
Ah, that's what you meant... Makes more sense, thanks. I fixed it.